tests/org.eclipse.reddeer.junit.test/src/org/eclipse/reddeer/junit/test/requirement/matcher/RequirementMatcherTest.java

```diff
@@ -15,148 +15,148 @@
 
 public class RequirementMatcherTest {
 
-	private static SimpleConfiguration config;
+	private static SimpleRequirement simpleReq;
+	private static SimpleConfiguration simpleConfig;
 	
+	private static ComplexRequirement complexReq;
 	private static ComplexConfiguration complexConfig;
 
 	@BeforeClass
 	public static void setup() {
-		config = new SimpleConfiguration();
-		config.setName("SimpleConfig");
-		config.setVersion("1.1");
+		simpleReq = new SimpleRequirement();
 		
+		simpleConfig = new SimpleConfiguration();
+		simpleConfig.setName("SimpleConfig");
+		simpleConfig.setVersion("1.1");
+		simpleReq.setConfiguration(simpleConfig);
+		
+		complexReq = new ComplexRequirement();
 		complexConfig = new ComplexConfiguration();
 		complexConfig.setComplexName("ComplexConfig");
 		complexConfig.setComplexType("ComplexType");
 		complexConfig.setComplexVersion("2.0");
-		complexConfig.setSimpleConfiguration(config);
+		complexConfig.setSimpleConfiguration(simpleConfig);
+		complexReq.setConfiguration(complexConfig);
 	}
 
 	@Test
 	public void testNestedAttributeOnPreciseMatch() {
-		RequirementRegexMatcher matcher = new RequirementRegexMatcher(ComplexConfiguration.class, "simpleConfiguration.name", "SimpleConfig");
+		RequirementMatcher matcher = new RequirementMatcher(ComplexReq.class, new RequirementRegexMatcher("simpleConfiguration.name", "SimpleConfig"));
 		assertTrue("Name attribute of configuration should be matched on precise match", matcher.matches(complexConfig));
 	}
 	
 	@Test
 	public void testNestedAttributeOnVersionMatch() {
-		RequirementVersionMatcher matcher = new RequirementVersionMatcher(ComplexConfiguration.class, "simpleConfiguration.version", ">1.0");
+		RequirementMatcher matcher = new RequirementMatcher(ComplexReq.class, new RequirementVersionMatcher("simpleConfiguration.version", ">1.0"));
 		assertTrue("Version attribute of configuration should be matched on version match", matcher.matches(complexConfig));
 	}
 	
 	@Test(expected=RequirementException.class)
 	public void testThrowExceptionOnNonexistingAttribute() {
-		RequirementVersionMatcher matcher = new RequirementVersionMatcher(ComplexConfiguration.class, "noConfig.version", ">1.0");
+		RequirementMatcher matcher = new RequirementMatcher(ComplexReq.class, new RequirementVersionMatcher("noConfig.version", ">1.0"));
 		matcher.matches(complexConfig);
 	}
 	
 	@Test(expected=RequirementException.class)
 	public void testThrowExceptionOnNonexistingNestedAttribute() {
-		RequirementVersionMatcher matcher = new RequirementVersionMatcher(ComplexConfiguration.class, "simpleConfiguration.what", ">1.0");
+		RequirementMatcher matcher = new RequirementMatcher(ComplexReq.class, new RequirementVersionMatcher("simpleConfiguration.what", ">1.0"));
 		matcher.matches(complexConfig);
 	}
 	
 	@Test
-	public void testRequirementRegexMatcherOnPreciseMatch() {
-		RequirementRegexMatcher matcher = new RequirementRegexMatcher(SimpleConfiguration.class, "name", "SimpleConfig");
```

Comment:
the logic for checking the correct class was moved from matchers to https://github.com/rawagner/reddeer/blob/984c47c70257752f4a1c9df4d696f559e804c050/plugins/org.eclipse.reddeer.junit/src/org/eclipse/reddeer/junit/internal/requirement/RequirementHelper.java#L102